lib/matplotlib/backends/backend_svg.py

```diff
@@ -813,75 +815,57 @@ def draw_image(self, gc, x, y, im, dx=None, dy=None, transform=None):
             # to the clip-path
             self.writer.start('g', attrib={'clip-path': 'url(#%s)' % clipid})
 
-        trans = [1,0,0,1,0,0]
-        if rcParams['svg.image_noscale']:
-            trans = list(im.get_matrix())
-            trans[5] = -trans[5]
-            attrib['transform'] = generate_transform([('matrix', tuple(trans))])
-            assert trans[1] == 0
-            assert trans[2] == 0
-            numrows, numcols = im.get_size()
-            im.reset_matrix()
-            im.set_interpolation(0)
-            im.resize(numcols, numrows)
-
-        h,w = im.get_size_out()
-
-        if dx is None:
-            w = 72.0*w/self.image_dpi
-        else:
-            w = dx
-
-        if dy is None:
-            h = 72.0*h/self.image_dpi
-        else:
-            h = dy
-
-        oid = getattr(im, '_gid', None)
-        url = getattr(im, '_url', None)
+        oid = gc.get_gid()
+        url = gc.get_url()
         if url is not None:
             self.writer.start('a', attrib={'xlink:href': url})
         if rcParams['svg.image_inline']:
             bytesio = io.BytesIO()
-            _png.write_png(np.array(im)[::-1], bytesio)
+            _png.write_png(im, bytesio)
```

Comment:
The transformation now takes care of that.  It's more efficient to do that, since the image will be flipped as it's resampled, rather than copying the data when `PyArray_ContiguousFromAny` is invoked and then resampling.